src/bindings/c/include/openvino/c/ov_property.h

```diff
@@ -0,0 +1,150 @@
+                       //!< every device accepts and turns into device-specific settings detail see
+                       //!< ov_performance_mode_e to get its hint's key name
+    NETWORK_NAME,              //!<  Read-only property<char *> to get a name of name of a model
+    INFERENCE_PRECISION_HINT,  //!< Read-write property<ov_element_type_e> to set the hint for device to use specified
+                               //!< precision for inference
+    OPTIMAL_BATCH_SIZE,  //!<  Read-only property<uint32_t> to query information optimal batch size for the given device
+                         //!<  and the network
+    MAX_BATCH_SIZE,  //!<  Read-only property to get maximum batch size which does not cause performance degradation due
+                     //!<  to memory swap impact.
+    PERFORMANCE_HINT_NUM_REQUESTS,  //!<  (Optional) property<uint32_t> that backs the Performance Hints by giving
+                                    //!<  additional information on how many inference requests the application will be
+                                    //!<  keeping in flight usually this value comes from the actual use-case  (e.g.
+                                    //!<  number of video-cameras, or other sources of inputs)
+} ov_property_key_e;
+
+/**
+ * @enum ov_property_value_type_e
+ * @brief Enum to define property value type.
+ */
+typedef enum {
+    BOOL = 0U,  //!< boolean data
+    CHAR,       //!< char data
+    ENUM,       //!< enum data
+    INT32,      //!< int32 data
+    UINT32,     //!< uint32 data
+    FLOAT,      //!< float data
+} ov_property_value_type_e;
+
+/**
+ * @struct ov_property_value_t
+ * @brief Represent a property value
+ */
+typedef struct {
+    void* ptr;
+    size_t cnt;
+    ov_property_value_type_e type;
+} ov_property_value_t;
+
+// Property
+/**
+ * @defgroup Property Property
+ * @ingroup openvino_c
+ * Set of functions representing of Property.
+ * @{
+ */
+
+/**
+ * @brief Create a property object.
+ * @ingroup property
+ * @param ov_status_e a status code, return OK if successful
+ */
+OPENVINO_C_API(ov_status_e) ov_property_create(ov_property_t** property);
+
+/**
+ * @brief Free property object.
+ * @ingroup property
+ * @param property will be released.
+ */
+OPENVINO_C_API(void) ov_property_free(ov_property_t* property);
+
+/**
+ * @brief Create a property value object.
+ * @ingroup property
+ * @param ov_status_e a status code, return OK if successful
+ */
+OPENVINO_C_API(ov_status_e) ov_property_value_create(ov_property_value_t** value);
+
+/**
+ * @brief Clean property data.
+ * @ingroup property
+ * @param property data will be clean.
+ */
+OPENVINO_C_API(void) ov_property_value_clean(ov_property_value_t* value);
+
+/**
+ * @brief Put <key, value> into property object.
+ * @ingroup property
+ * @param property will be add new <key, value>.
+ */
+OPENVINO_C_API(ov_status_e) ov_property_put(ov_property_t* property, ov_property_key_e key, ov_property_value_t* value);
```

Comment:
can we provide examples how to pass several properties to compile model?

Reply:
Original definition support property list, and current property have become very complex and so not support such property list, maybe we can add it later.